File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class RobotNewsStationNode : public rclcpp::Node
{
public:
    RobotNewsStationNode() : Node("robot_news_station"), robot_name_("Lichi01")
    {
        Publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        timer_ = this->create_wall_timer(0.5s, std::bind(&RobotNewsStationNode::PublishNews, this));
        RCLCPP_INFO(this->get_logger(), "Robot News station has been started");
    }

private:
    void PublishNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Namaste this is ") + robot_name_ + 
        std::string(", do you copy?");
        Publisher_->publish(msg);
    }

    std::string robot_name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr Publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class robot_News_Station_Node (Node):
        def __init__(self):
            super().__init__("robot_news_station")
            self.robot_name_ = "Lichi01"
            self.publisher_ = self.create_publisher(String, "robot_news", 10)
            self.timer_ = self.create_timer(0.5, self.publish_news)
            self.get_logger().info("Robot news station has been started")

        def publish_news(self):
             msg = String()
             msg.data = "Namaste! I am " + self.robot_name_+ ", how are you? do you copy?"
             self.publisher_.publish(msg)



def main(args=None):
    rclpy.init(args=args)
    node = robot_News_Station_Node()
    rclpy.spin(node)()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()
